lib/system_deps.js

```diff
@@ -0,0 +1,44 @@
+var childProcess = require('child_process')
+var path = require('path');
+
+function cmdFound(cmd) {
+  return (!childProcess.spawnSync(cmd, {shell:true}).stderr.length);
+}
```

Comment:
stderr output doesn't mean the command is not found. Let's do this flat and not with nested `if`s like so:
```
// validate php
// childProcess.spawnSync('php --version') and check for expected output

// validate composer
// childProcess.spawnSync('composer --version') and check for expected output

if (!phpPresent) {
 throw new Error('php runtime cannot be found');
}

...
```